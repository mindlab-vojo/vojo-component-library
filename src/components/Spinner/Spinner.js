import React from 'react'
import PropTypes from 'prop-types'

import * as Spinner from './Spinner.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'

const SpinnerComponent = ({
  height,
  spinnerArea,
  width,
}) => {
  const renderComponent = (theme) => (
    <Spinner.Wrapper
      clasname="Spinner"
      spinnerArea={spinnerArea}>
      <Spinner.Spinner
        version="1.1" 
        xmlns="http://www.w3.org/2000/svg" 
        viewBox="0 0 512 512"
        xmlSpace="preserve"
        height={height} 
        width={width}>
        <g>
          <path 
            fill={theme.colors.primaryColor}
            d="M266.3,0c-34.3,0-62.1,27.8-62.1,62.1s27.8,62.1,62.1,62.1s62.1-27.8,62.1-62.1S300.6,0,266.3,0z M120.2,71.1
		c-28.5,0-51.7,23.2-51.7,51.7s23.2,51.7,51.7,51.7s51.7-23.2,51.7-51.7S148.8,71.1,120.2,71.1z M412.4,102.1
		c-11.4,0-20.7,9.3-20.7,20.7s9.3,20.7,20.7,20.7c11.4,0,20.7-9.3,20.7-20.7S423.8,102.1,412.4,102.1z M59.5,222.4
		c-25.7,0-46.5,20.8-46.5,46.5s20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5S85.2,222.4,59.5,222.4z M473.2,243.1
		c-14.3,0-25.9,11.6-25.9,25.9c0,14.3,11.6,25.9,25.9,25.9c14.3,0,25.9-11.6,25.9-25.9C499.1,254.6,487.5,243.1,473.2,243.1z
		M120.2,373.7c-22.9,0-41.4,18.5-41.4,41.4s18.5,41.4,41.4,41.4s41.4-18.5,41.4-41.4S143.1,373.7,120.2,373.7z M412.4,384
		c-17.1,0-31,13.9-31,31s13.9,31,31,31c17.1,0,31-13.9,31-31S429.6,384,412.4,384z M266.3,439.6c-20,0-36.2,16.2-36.2,36.2
		s16.2,36.2,36.2,36.2s36.2-16.2,36.2-36.2S286.3,439.6,266.3,439.6z"/>
        </g>
      </Spinner.Spinner>
    </Spinner.Wrapper>
  )
  
  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

SpinnerComponent.propTypes = {
  height: PropTypes.string,
  spinnerArea: PropTypes.string,
  width: PropTypes.string,
}

SpinnerComponent.defaultProps = {
  height: '50px',
  spinnerArea: '100px',
  width: '50px',
}

export default SpinnerComponent
