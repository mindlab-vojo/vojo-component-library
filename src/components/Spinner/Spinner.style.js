import styled, { keyframes } from 'styled-components'

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;
export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  height: ${props => props.spinnerArea ? props.spinnerArea : '100px'};
  justify-content: center
`
export const Spinner = styled.svg`
  animation: ${rotate} 2s linear infinite;
  enable-background:new 0 0 512 512;
`