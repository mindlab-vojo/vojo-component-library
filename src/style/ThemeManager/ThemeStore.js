import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore