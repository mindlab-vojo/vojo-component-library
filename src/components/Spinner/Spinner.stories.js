import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';

import Spinner from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

import Container from '../Container';

export const component = () => {  
  const height = text('height', '50px')
  const spinnerArea = text('spinnerArea', '100px')
  const width = text('width', '50px')

  return (
    <Container>
      <Spinner 
        height={height}
        spinnerArea={spinnerArea}
        width={width}/>
    </Container>
  )
}

export default {
  title: categories.Spinners,
  component: component,
  decorators: [withKnobs, withA11y, (story) => <ThemeProvider>{ story() }</ThemeProvider>]
}

component.story = {
  name: 'Spinner',
};
  