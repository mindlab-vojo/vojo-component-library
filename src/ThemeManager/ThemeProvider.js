import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer