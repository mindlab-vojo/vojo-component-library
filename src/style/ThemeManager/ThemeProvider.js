import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer