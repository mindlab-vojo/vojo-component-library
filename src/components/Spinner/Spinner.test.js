import React from 'react';
import { render } from '@testing-library/react';

import Spinner from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test Spinner component render', () => {
  const component = render(
    <ThemeProvider>
      <Spinner/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
