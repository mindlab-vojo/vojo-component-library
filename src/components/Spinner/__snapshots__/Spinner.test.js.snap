// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Spinner component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm jOAJCI"
      >
        <svg
          class="sc-AxirZ iUBOPB"
          height="50px"
          version="1.1"
          viewBox="0 0 512 512"
          width="50px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M266.3,0c-34.3,0-62.1,27.8-62.1,62.1s27.8,62.1,62.1,62.1s62.1-27.8,62.1-62.1S300.6,0,266.3,0z M120.2,71.1 c-28.5,0-51.7,23.2-51.7,51.7s23.2,51.7,51.7,51.7s51.7-23.2,51.7-51.7S148.8,71.1,120.2,71.1z M412.4,102.1 c-11.4,0-20.7,9.3-20.7,20.7s9.3,20.7,20.7,20.7c11.4,0,20.7-9.3,20.7-20.7S423.8,102.1,412.4,102.1z M59.5,222.4 c-25.7,0-46.5,20.8-46.5,46.5s20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5S85.2,222.4,59.5,222.4z M473.2,243.1 c-14.3,0-25.9,11.6-25.9,25.9c0,14.3,11.6,25.9,25.9,25.9c14.3,0,25.9-11.6,25.9-25.9C499.1,254.6,487.5,243.1,473.2,243.1z M120.2,373.7c-22.9,0-41.4,18.5-41.4,41.4s18.5,41.4,41.4,41.4s41.4-18.5,41.4-41.4S143.1,373.7,120.2,373.7z M412.4,384 c-17.1,0-31,13.9-31,31s13.9,31,31,31c17.1,0,31-13.9,31-31S429.6,384,412.4,384z M266.3,439.6c-20,0-36.2,16.2-36.2,36.2 s16.2,36.2,36.2,36.2s36.2-16.2,36.2-36.2S286.3,439.6,266.3,439.6z"
              fill="#391DDD"
            />
          </g>
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm jOAJCI"
    >
      <svg
        class="sc-AxirZ iUBOPB"
        height="50px"
        version="1.1"
        viewBox="0 0 512 512"
        width="50px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M266.3,0c-34.3,0-62.1,27.8-62.1,62.1s27.8,62.1,62.1,62.1s62.1-27.8,62.1-62.1S300.6,0,266.3,0z M120.2,71.1 c-28.5,0-51.7,23.2-51.7,51.7s23.2,51.7,51.7,51.7s51.7-23.2,51.7-51.7S148.8,71.1,120.2,71.1z M412.4,102.1 c-11.4,0-20.7,9.3-20.7,20.7s9.3,20.7,20.7,20.7c11.4,0,20.7-9.3,20.7-20.7S423.8,102.1,412.4,102.1z M59.5,222.4 c-25.7,0-46.5,20.8-46.5,46.5s20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5S85.2,222.4,59.5,222.4z M473.2,243.1 c-14.3,0-25.9,11.6-25.9,25.9c0,14.3,11.6,25.9,25.9,25.9c14.3,0,25.9-11.6,25.9-25.9C499.1,254.6,487.5,243.1,473.2,243.1z M120.2,373.7c-22.9,0-41.4,18.5-41.4,41.4s18.5,41.4,41.4,41.4s41.4-18.5,41.4-41.4S143.1,373.7,120.2,373.7z M412.4,384 c-17.1,0-31,13.9-31,31s13.9,31,31,31c17.1,0,31-13.9,31-31S429.6,384,412.4,384z M266.3,439.6c-20,0-36.2,16.2-36.2,36.2 s16.2,36.2,36.2,36.2s36.2-16.2,36.2-36.2S286.3,439.6,266.3,439.6z"
            fill="#391DDD"
          />
        </g>
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
